ComfySoda_Unity2018.2.7f1/Assets/Scripts/ComySodaEngine/EnemyController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Unity;
+using UnityEditor;
+using UnityEngine;
+
+
+namespace cs
+{
+    public class EnemyController : MonoBehaviour
+    {
+
+        #region Static referencing
+        public static EnemyController enemyController = null;
+        private void Awake()
+        {
+            Debug.Assert(enemyController == null, "Error: There should only be one enemy controller assigned to \"EnemyController.enemyController\" at a time");
+            enemyController = this;
+        }
+
+        private void OnDestroy()
+        {
+            Debug.Assert(enemyController == this, "Error: The enemy controller assigned to \"EnemyController.enemyController\" does not equal: " + gameObject);
+            enemyController = null;
+        }
+        #endregion
+
+
+        #region Enemy distance
+        [SerializeField]
+        public float CLOSE = 7;
+        [SerializeField]
+        public float MID = 12;
+
+        public enum EnemyRange { CLOSE, MID, FAR};
+        public EnemyRange enemyRange = EnemyRange.FAR;
+        public List<EnemyLogic> enemyList = EnemyLogic.enemyList;
+
+        void calculateAndAssignEnemyRange()
+        {
+            foreach(EnemyLogic enemyLogic in enemyList)
+            {
+                enemyRange = EnemyRange.FAR;
+                float tempDistance = Vector2.Distance(PlayerLogic.playerLogic.transform.position, enemyLogic.transform.position);
+                {
+                    if(tempDistance <= CLOSE || enemyRange==EnemyRange.CLOSE)
+                    {
+                        enemyRange = EnemyRange.CLOSE;
+                    }
+                    else if(tempDistance <= MID || enemyRange == EnemyRange.MID)
+                    {
+                        enemyRange = EnemyRange.MID;
+                    }
+                }
+            }
+        }
+        #endregion
```

Comment:
I don't think this code works the way you expect it to. This code only applies closeness logic to the last enemy it checks. You'll need to move your EnemRange.Far initialization out of the loop. Also, what's up with the random scoping around the if/else block?

Reply:
You're right, fixed.  and scoping... oh, LOL.  I think my tab spacing got off somehow when I was trying to figure out if I could do it with a nested loop and I just glanced through it and decided float tempDistance was a function in my head.  I'll remove those, they don't need to be there, haha.